### PR TITLE
[CAT-92] & [CAT-71] Document Report and Submission Retry

### PR DESCRIPTION
**Pull request by goatrocks**
## Summary

Adds the document report metrics and also the submission retry query.

## Changelog (optional)
- Created new metrics query `GetDocumentReport`
- Created new submission mutation `SubmissionRetry`

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:
A quick checklist of things that should be done before the code is opened to review by others on the team.
Delete any that are not applicable.

- [x] Jira story/task is put into PR title
- [x] Code has been self-reviewed
- [x] Hard-to-understand areas have been commented
- [x] Documentation has been updated
- [x] My changes generate no new warnings
- [x] Unit tests have been added for base functionality and known edge cases

### CODE REVIEW

File: indico/queries/submission.py [hunk 1]
```diff
@@ -399,3 +399,43 @@ def __init__(
     def process_response(self, response) -> Job:
         response = super().process_response(response)["submitAutoReview"]
         return Job(id=response["jobId"])
+
+
+class RetrySubmission(GraphQLRequest):
+    """
+    Given a list of submission ids, retry those failed submissions.
```
Comment (Scott771) [inline indico/queries/submission.py:406]: does this check that the IDs past to it actually failed?

Comment (goatrocks) [reply indico/queries/submission.py:406]: The backend does check and returns an error string if the ID is already done

**Review by jonaskaz — COMMENTED** [review]
looks good
